src/lib/paginator/paginator.ts

```diff
@@ -96,6 +96,9 @@ export class MatPaginator implements OnInit, OnDestroy {
   }
   private _pageSizeOptions: number[] = [];
 
+  /** Whether to show the page size selection UI to the user. */
+  @Input() pageSizeEnabled = true;
```

Comment:
@andrewseguin I was torn between doing this or having some logic where the select is hidden if you pass in an empty array of options. I went with this since it's a bit more obvious what is going on.

Reply:
At the moment an empty array will display a label (whereas if there are several elements, it will display a dropdown) so I think overwriting the current behaviour wouldn't have been a good idea. 

@crisbeto btw, thanks for the fast response to the issue.